File: xwui/src/components/super/XWUIScriptEditor/testers/files/sample.go
// Sample Go Code
// Comprehensive Go example demonstrating various features

package main

import (
	"fmt"
	"time"
)

// Metadata represents document metadata
type Metadata struct {
	Author  string
	Created string
	Tags    []string
}

// Item represents a document item
type Item struct {
	ID         int
	Name       string
	Active     bool
	Price      float64
	Categories []string
}

// Document represents the main document
type Document struct {
	Name        string
	Version     string
	Description string
	Metadata    Metadata
	Items       []Item
}

// NewDocument creates a new document with default metadata
func NewDocument(name, version, description string) *Document {
	return &Document{
		Name:        name,
		Version:     version,
		Description: description,
		Metadata: Metadata{
			Author:  "XWUI Team",
			Created: time.Now().Format(time.RFC3339),
			Tags:    []string{"example", "go"},
		},
		Items: make([]Item, 0),
	}
}

// AddItem adds an item to the document
func (d *Document) AddItem(item Item) {
	d.Items = append(d.Items, item)
}

// GetActiveItems returns all active items
func (d *Document) GetActiveItems() []Item {
	var activeItems []Item
	for _, item := range d.Items {
		if item.Active {
			activeItems = append(activeItems, item)
		}
	}
	return activeItems
}

// GetTotalPrice calculates the total price of active items
func (d *Document) GetTotalPrice() float64 {
	total := 0.0
	for _, item := range d.Items {
		if item.Active {
			total += item.Price
		}
	}
	return total
}

// String returns a string representation of the document
func (d *Document) String() string {
	return fmt.Sprintf("Document{Name: %s, Version: %s, Items: %d}",
		d.Name, d.Version, len(d.Items))
}

func main() {
	// Create document
	doc := NewDocument(
		"Sample Go Document",
		"1.0.0",
		"Go code example",
	)
	
	// Add items
	doc.AddItem(Item{
		ID:         1,
		Name:       "First Item",
		Active:     true,
		Price:      29.99,
		Categories: []string{"electronics", "gadgets"},
	})
	
	doc.AddItem(Item{
		ID:         2,
		Name:       "Second Item",
		Active:     false,
		Price:      49.99,
		Categories: []string{"books", "education"},
	})
	
	// Use methods
	fmt.Printf("Document: %s\n", doc.Name)
	fmt.Printf("Active items: %d\n", len(doc.GetActiveItems()))
	fmt.Printf("Total price: $%.2f\n", doc.GetTotalPrice())
	fmt.Printf("Document: %s\n", doc.String())
}

File: xwui/src/components/super/XWUIScriptEditor/testers/files/sample.c
// C Example
// Comprehensive C code demonstrating various features

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

// Metadata structure
typedef struct {
    char author[100];
    char created[50];
    char tags[200];
} Metadata;

// Item structure
typedef struct {
    int id;
    char name[100];
    bool active;
    double price;
    char categories[200];
} Item;

// Document structure
typedef struct {
    char name[100];
    char version[20];
    char description[200];
    Metadata metadata;
    Item items[10];
    int item_count;
} Document;

// Function to initialize document
void initDocument(Document* doc, const char* name, const char* version, const char* description) {
    strcpy(doc->name, name);
    strcpy(doc->version, version);
    strcpy(doc->description, description);
    strcpy(doc->metadata.author, "XWUI Team");
    strcpy(doc->metadata.created, "2025-01-15T10:30:00Z");
    strcpy(doc->metadata.tags, "example,c,c-language");
    doc->item_count = 0;
}

// Function to add item
void addItem(Document* doc, int id, const char* name, bool active, double price) {
    if (doc->item_count < 10) {
        Item* item = &doc->items[doc->item_count];
        item->id = id;
        strcpy(item->name, name);
        item->active = active;
        item->price = price;
        strcpy(item->categories, "electronics,gadgets");
        doc->item_count++;
    }
}

// Function to get total price of active items
double getTotalPrice(Document* doc) {
    double total = 0.0;
    for (int i = 0; i < doc->item_count; i++) {
        if (doc->items[i].active) {
            total += doc->items[i].price;
        }
    }
    return total;
}

// Function to count active items
int countActiveItems(Document* doc) {
    int count = 0;
    for (int i = 0; i < doc->item_count; i++) {
        if (doc->items[i].active) {
            count++;
        }
    }
    return count;
}

// Main function
int main() {
    Document doc;
    initDocument(&doc, "Sample C Document", "1.0.0", "C code example");
    
    addItem(&doc, 1, "First Item", true, 29.99);
    addItem(&doc, 2, "Second Item", false, 49.99);
    
    printf("Document: %s\n", doc.name);
    printf("Version: %s\n", doc.version);
    printf("Active items: %d\n", countActiveItems(&doc));
    printf("Total price: $%.2f\n", getTotalPrice(&doc));
    
    return 0;
}

